### PR TITLE
Bug: no validation and confirmation on comments

### LINKED ISSUE #2277: No validation and confirmation on comments.
**Describe the bug**
Users are able to comment blank texts.

**To Reproduce**

1. Enter the comments view.
2. Without typing anything, hit 'send'.

**Expected behavior**
1. Should not allow sending the comment if the comment is empty.
2. Should provide confirmation once comment is successfully uploaded.

**Actual behavior**
Clicking 'send' is calling the API with empty comment message and adding the blank comment locally.
After an error is received from the backend, the comment disappears.
This leads to a poor user experience.

Can I work on this issue?

### PR DESCRIPTION
**Pull request by Dante291**
**What kind of change does this PR introduce?**

This PR introduces an enhancement by implementing validation and confirmation mechanisms for comments.

**Issue Number:**

Fixes #2277 

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

[validation.webm](https://github.com/PalisadoesFoundation/talawa/assets/103507835/a5fa34a6-d6f4-47ff-81f3-a51a597c2d99)

**If relevant, did you update the documentation?**

No

**Does this PR introduce a breaking change?**

No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by Cioppolo14** [comment]
Please fix the failing tests.

**Comment by Dante291** [comment]
@palisadoes @Cioppolo14 please assign reviewers.

**Comment by Cioppolo14** [comment]
@literalEval @anwersayeed Can you review this PR?

**Comment by Dante291** [comment]
@noman2002 @literalEval please review this

File: lib/views/after_auth_screens/feed/individual_post.dart [hunk 1]
```diff
@@ -59,14 +70,32 @@ class _IndividualPostViewState extends State<IndividualPostView> {
...
-              },
+              key: const Key('sendButton'),
+              style: _isCommentValid == false
+                  ? ButtonStyle(
+                      overlayColor:
+                          MaterialStateProperty.all(Colors.transparent),
+                    )
+                  : null,
+
+              ///check if button is enabled when comment is valid
```
Comment (literalEval) [inline lib/views/after_auth_screens/feed/individual_post.dart:81]: Use `//` for comments.

File: lib/views/after_auth_screens/feed/individual_post.dart [hunk 2]
```diff
@@ -104,6 +133,14 @@ class _IndividualPostViewState extends State<IndividualPostView> {
...
+///
+/// **params**:
+/// * `context`: Context for padding method
+/// * `text`: text on which padding should be applied
+///
+/// **returns**:
+/// * `Padding`: padding for the text
 Padding buildPadding(BuildContext context, String text) {
   return Padding(
     padding: const EdgeInsets.symmetric(vertical: 8.0),
```
Comment (literalEval) [inline lib/views/after_auth_screens/feed/individual_post.dart:146]: Use dynamic sizing instead.

File: lib/views/after_auth_screens/feed/individual_post.dart [hunk 3]
```diff
@@ -121,6 +158,7 @@ class IndividualPageLikeSection extends StatelessWidget {
     required this.usersLiked,
   });
 
+  /// List of user who liked the post.
```
Comment (literalEval) [inline lib/views/after_auth_screens/feed/individual_post.dart:161]: *users

File: lib/views/after_auth_screens/feed/individual_post.dart [hunk 4]
```diff
@@ -154,7 +192,11 @@ class IndividualPostCommentSection extends StatelessWidget {
     required this.comments,
     required this.postID,
   });
+
+  /// LIst of comments on a post.
```
Comment (literalEval) [inline lib/views/after_auth_screens/feed/individual_post.dart:196]: *List

File: lib/views/after_auth_screens/feed/individual_post.dart [hunk 5]
```diff
@@ -228,7 +271,13 @@ class CommentTemplate extends StatelessWidget {
...
 }
 
-/// likedUserCircleAvatar returns a widget of the individual user liked the post.
+/// Circle avatar of users who have liked the post.
+///
+/// **params**:
+/// * `user`: User who liked the post
+///
+/// **returns**:
+/// * `Widget`: returns the circle avatar of the likedBy user who liked the post
```
Comment (literalEval) [inline lib/views/after_auth_screens/feed/individual_post.dart:280]: Don't use words like returns, takes etc. It is obvious. Moreover, make the comment multi-line if the single line goes too wide.

**Review by literalEval — CHANGES_REQUESTED** [review]

**Comment by Dante291** [comment]
@literalEval made requested changes

**Review by literalEval — CHANGES_REQUESTED** [review]
There are some guidelines which you should try on enforcing yourself, like:

1. Writing better comments.
2. Adding trailing commas and refactoring.
3. Adding proper spacing on comments and docs.

**Comment by Dante291** [comment]
@literalEval 

1. I have revisited and have added more meaningful docs in the code to provide better clarity.
2.I have added trailing commas wherever necessary for proper readability of code.
3. Added proper spacing on comments and docs.

If, after these updates, there are still areas that require improvement, please feel free to specify the errors or provide additional guidance. Your feedback is highly valued, and I'm committed to delivering code that meets the highest standards.

**Review by noman2002 — APPROVED** [review]
LGTM

**Review by palisadoes — APPROVED** [review]

**Pull request merged**